Added some helpers for string manipulation.

SimonPStevens opened the pull request:
**Parent issue** 

Closes https://github.com/mockoon/mockoon/issues/413

**Technical implementation details**

Added some string manipulation helpers that just call the equivalent Javascript methods directly.

255kb commented:
Thank you @SimonPStevens for these helpers :) 
Like for the `dateTimeShift` helper, you should account for the case where no params are passed. We currently have an error with the `indexOf` (didn't test the other helpers) when used like this: `{{indexOf}}`. 
Could you please also add more test cases? Like for the `setVar`: one case without param, with only the first param, with string param, number param, another embedded helper (`indexOf (body 'prop') 'a'`), etc. Helpers code is a little bit easy to break and I want it to be covered as much as possible. 
Thanks!

SimonPStevens commented:
Yeah, no prob. I'll get to this later in the week.

SimonPStevens commented:
Hey. Sorry it took me way longer than I said to get round to doing this. I've added some more unit tests, and fixed the error when the wrong parameter type is passed. Let me know if there is anything else that needs doing.

SimonPStevens commented:
Actually. Hold off. This still doesn't work with variables, and I'm not quite sure why. I'm still working on it.

SimonPStevens commented:
OK. I think this is good now.

That opened up a bit of a can of worms with the different parameter types that can be passed to the helpers. I've covered everything I can think of now, but if you can see anything I've missed then let me know.

255kb commented on src/libs/template-parser.ts:
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
+          : Number(position.toString());
+
+      if (typeof position === 'number') {
+        return data.toString().indexOf(search.toString(), position);
```
`data` and `search` are already converted to string above
```suggestion
        return data.indexOf(search, position);
```

255kb commented on src/libs/template-parser.ts:
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
+      if (typeof position === 'number') {
+        return data.toString().indexOf(search.toString(), position);
+      } else {
+        return data.toString().indexOf(search.toString());
```
`data` and `search` are already converted to string above
```suggestion
        return data.indexOf(search);
```

255kb commented on src/libs/template-parser.ts:
```diff
@@ -335,6 +335,83 @@ const TemplateParserHelpers = function (request: Request) {
+          ? ''
+          : search.toString();
+
+      return data.toString().includes(search.toString());
```
`data` and `search` are already converted to string above
```suggestion
      return data.includes(search);
```

255kb commented:
@SimonPStevens It looks good. Just 3 `toString()` that seems redundant to me. Then it's ready to be merged!

SimonPStevens commented:
@255kb All done.

255kb commented:
Thanks @SimonPStevens !
